src/main/java/org/mitre/synthea/export/rif/BB2RIFExporter.java

```diff
@@ -0,0 +1,359 @@
+    endState.setProperty("exporter.bfd.pde_id_start", RIFExporter.nextPdeId.toString());
+    endState.setProperty("exporter.bfd.mbi_start", RIFExporter.nextMbi.toString());
+    endState.setProperty("exporter.bfd.hicn_start", RIFExporter.nextHicn.toString());
+    endState.setProperty("exporter.bfd.fi_doc_cntl_num_start",
+            RIFExporter.nextFiDocCntlNum.toString());
+    endState.setProperty("exporter.bfd.carr_clm_cntl_num_start",
+            RIFExporter.nextCarrClmCntlNum.toString());
+    File outputDir = Exporter.getOutputFolder("bfd", null);
+    FileOutputStream f = new FileOutputStream(new File(outputDir, "end_state.properties"));
+    endState.store(f, "BFD Properties End State");
+    f.close();
+  }
+
+  /**
+   * Display DME codes that were not mappable during export.
+   * These missing codes might be accidental, they may be intentional.
+   */
+  public void displayMissingDmeCodes() {
+    String description;
+    for (String code : missingDmeCodes.keySet()) {
+      description = missingDmeCodes.get(code);
+      System.err.println(" *** Possibly Missing DME Code: " + code + " | " + description);
+    }
+  }
+
+  /**
+   * Export a single person.
+   * @param person the person to export
+   * @param stopTime end time of simulation
+   * @param yearsOfHistory number of years of claims to export
+   * @throws IOException if something goes wrong
+   */
+  public void export(Person person, long stopTime, int yearsOfHistory) throws IOException {
+    Map<EXPORT_SUMMARY, String> exportCounts = new HashMap<>();
+    long startTime = stopTime - Utilities.convertTime("years", yearsOfHistory);
+    BeneficiaryExporter beneExp = new BeneficiaryExporter(startTime, stopTime, this);
+    exportCounts.put(EXPORT_SUMMARY.BENE_ID, beneExp.export(person));
+    beneExp.exportHistory(person);
+    exportCounts.put(EXPORT_SUMMARY.INPATIENT_CLAIMS,
+            Long.toString(new InpatientExporter(startTime, stopTime, this).export(person)));
+    exportCounts.put(EXPORT_SUMMARY.OUTPATIENT_CLAIMS,
+            Long.toString(new OutpatientExporter(startTime, stopTime, this).export(person)));
+    exportCounts.put(EXPORT_SUMMARY.CARRIER_CLAIMS,
+            Long.toString(new CarrierExporter(startTime, stopTime, this).export(person)));
+    exportCounts.put(EXPORT_SUMMARY.PDE_CLAIMS,
+            Long.toString(new PDEExporter(startTime, stopTime, this).export(person)));
+    exportCounts.put(EXPORT_SUMMARY.DME_CLAIMS,
+            Long.toString(new DMEExporter(startTime, stopTime, this).export(person)));
+    exportCounts.put(EXPORT_SUMMARY.HHA_CLAIMS,
+            Long.toString(new HHAExporter(startTime, stopTime, this).export(person)));
+    exportCounts.put(EXPORT_SUMMARY.HOSPICE_CLAIMS,
+            Long.toString(new HospiceExporter(startTime, stopTime, this).export(person)));
+    exportCounts.put(EXPORT_SUMMARY.SNF_CLAIMS,
+            Long.toString(new SNFExporter(startTime, stopTime, this).export(person)));
```

Comment:
Classically (potentially still) the most expensive operation in Java was `new` Object creation...

We'd probably get better performance at scale if you create the `*Exporter` classes once, and just reuse them.

Reply:
Fixed.